vtex/faststore
# Feat/add increment address feature

## What's the purpose of this pull request?

Completing address information via VTEX's native address database API.

## How it works?

The store must configure whether or not it wants to use the API to complement Postal Code addresses.
Some countries such as France do not natively have the address complement in the Order Form at active checkout and the address is incomplete in the Order Form.
https://github.com/vtex/address-form/blob/main/react/country/FRA.ts#L23

## How to test it?

Change the storeConfiguration and check the address incremention:

![image](https://github.com/vtex/faststore/assets/67066494/fa067a07-94d0-495d-8a42-abb15bda26a8)

![image](https://github.com/vtex/faststore/assets/67066494/2d00471b-d550-4291-8226-cc43578ce645)


For countries like USA and Brasil this process happens automatically for the checkout API so you should just validate that the incremental process will work with the console log.

You can use the 99501 to test.

You should expect at the console log when its true to be:

<img width="522" alt="image" src="https://github.com/vtex/faststore/assets/67066494/b9c744fa-ca44-4afe-a682-634e0b3168b8">

<img width="257" alt="image" src="https://github.com/vtex/faststore/assets/67066494/6dc6cd89-cfd5-4232-840e-b3b4c11fc1fe">

and when its false to be undefined:

<img width="241" alt="image" src="https://github.com/vtex/faststore/assets/67066494/5826a401-ac25-4396-b680-d8dcad63e265">

<img width="257" alt="image" src="https://github.com/vtex/faststore/assets/67066494/43cf48e3-71b3-496c-9d38-bb741e2561d7">
